cmd/manager/main.go

```diff
@@ -289,6 +305,21 @@ func main() {
 		os.Exit(1)
 	}
 
+	if globalPullSecret != "" {
+		secretParts := strings.Split(globalPullSecret, "/")
+		setupLog.Info("Creating SecretSyncer controller for watching secret", secretParts[1], "in namespace", secretParts[0])
+		err := (&controllers.SecretSyncerReconciler{
+			Client:          mgr.GetClient(),
+			AuthFilePath:    authFilePath,
+			SecretName:      secretParts[1],
+			SecretNamespace: secretParts[0],
```

Comment:
We need to configure the cache to start a secret informer in the specified secret's namespace. We should not have a cluster-wide secret infomer. That would require the controller to have cluster-wide list-watch permissions on secrets, and that's a big no-no.

Reply:
[So the only combination that seems to work](https://github.com/operator-framework/operator-controller/pull/1322/files#diff-435eecb1d2af40ee96747f88ab71eb2758da989c92f76dcb1f714fc1b300c633R158-R167) (controller actually receives events for and reconciles the object) is 

```
if globalPullSecretKey != nil {
		cacheOptions.ByObject[&corev1.Secret{}] = crcache.ByObject{
			Field: fields.SelectorFromSet(map[string]string{
				"metadata.name": globalPullSecretKey.Name,
			}),
		}
		cacheOptions.DefaultNamespaces[globalPullSecretKey.Namespace] = crcache.Config{
			LabelSelector: k8slabels.Everything(),
		}
}
```

ie 
* in `ByObject`: use field selector 
* in `Config`: use labels.Everything()

If I switch the field used in `Config` to `FieldSelector` and pass in 

```
fields.SelectorFromSet(map[string]string{
	"metadata.name": globalPullSecretKey.Name,
})
```

the controller doesn't reconcile. 

I'm guessing the combination I mentioned will do the job (cache Secrets with `metadata.Name`=<name-we-want>, with labels.Everything() from the namespace). I'm guessing because:what's a good way of checking what's being cached?

Reply:
We got it sorted out in the latest commit!